File: app/src/main/java/pe/com/gadolfolozano/kotlinzomatoapi/ui/util/RestaurantInfoWindow.java
package pe.com.gadolfolozano.kotlinzomatoapi.ui.util;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;
import com.bumptech.glide.Glide;
import com.google.android.gms.maps.GoogleMap;
import com.google.android.gms.maps.model.Marker;
import pe.com.gadolfolozano.kotlinzomatoapi.R;
import pe.com.gadolfolozano.kotlinzomatoapi.ui.model.RestaurantMarker;

public class RestaurantInfoWindow implements GoogleMap.InfoWindowAdapter {

    private View view;
    private Context context;

    public RestaurantInfoWindow(Context context) {
        this.context = context;

        LayoutInflater inflater = (LayoutInflater) this.context.getApplicationContext().getSystemService(Context.LAYOUT_INFLATER_SERVICE);

        view = inflater.inflate(R.layout.layout_info_window,
                null);
    }


    @Override
    public View getInfoWindow(Marker marker) {
        if (marker != null
                && marker.isInfoWindowShown()) {
            marker.hideInfoWindow();
            marker.showInfoWindow();
        }
        return null;
    }

    @Override
    public View getInfoContents(Marker marker) {
        final ImageView imgThumb = view.findViewById(R.id.img_thumb);
        final TextView txtTitle = view.findViewById(R.id.txt_title);
        final TextView txtDetail = view.findViewById(R.id.txt_detail);

        if (marker.getTag() instanceof RestaurantMarker) {
            RestaurantMarker restaurantMarker = (RestaurantMarker) marker.getTag();

            txtTitle.setText(restaurantMarker.getName());
            txtDetail.setText(restaurantMarker.getAddress());
            Glide.with(context).load(restaurantMarker.getThumb()).into(imgThumb);
        }

        return view;
    }
}

File: app/src/main/java/pe/com/gadolfolozano/kotlinzomatoapi/ui/util/RestaurantInfoWindow.kt
package pe.com.gadolfolozano.kotlinzomatoapi.ui.util

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.widget.ImageView
import android.widget.TextView
import com.bumptech.glide.Glide
import com.bumptech.glide.load.engine.DiskCacheStrategy
import com.google.android.gms.maps.GoogleMap
import com.google.android.gms.maps.model.Marker
import pe.com.gadolfolozano.kotlinzomatoapi.R
import pe.com.gadolfolozano.kotlinzomatoapi.ui.model.RestaurantMarker

class RestaurantInfoWindow(private val context: Context) : GoogleMap.InfoWindowAdapter {

    private val view: View
    val previousImageUrl: String? = null

    init {
        val inflater =
            this.context.applicationContext.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater

        view = inflater.inflate(R.layout.layout_info_window, null)
    }


    override fun getInfoWindow(marker: Marker?): View? {
        return null
    }

    override fun getInfoContents(marker: Marker): View {
        val imgThumb = view.findViewById<ImageView>(R.id.img_thumb)
        val txtTitle = view.findViewById<TextView>(R.id.txt_title)
        val txtDetail = view.findViewById<TextView>(R.id.txt_detail)

        if (marker.tag is RestaurantMarker) {
            val restaurantMarker = marker.tag as RestaurantMarker?

            txtTitle.text = restaurantMarker?.name
            txtDetail.text = restaurantMarker?.address
            if (restaurantMarker?.thumb != null) {
                imgThumb.visibility = View.VISIBLE
                Glide.with(context)
                    .load(restaurantMarker.thumb)
                    .diskCacheStrategy(DiskCacheStrategy.ALL)
                    .into(imgThumb)
            } else {
                imgThumb.visibility = View.GONE
            }
        }

        return view
    }
}
